docs: Consolidate and update HIP catch test build documentation

cjatin opened the pull request:
## Motivation

Consolidate and update HIP catch test documentation with current build steps, especially for standalone test compilation after the Catch2 v3 upgrade.

## Technical Details

- Remove obsolete `README-doc.md` and merge its content into `projects/hip-tests/README.md`
- Add detailed instructions for building standalone tests with Catch2 v3 (system-wide install and FetchContent)
- Document `amdclang++` as the recommended compiler with offload-arch usage
- Update `projects/hip-tests/catch/README.md` with macro documentation and restructured build docs

## JIRA ID

NA

## Test Plan

Documentation-only change; no tests required.

## Test Result

NA

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

cjatin committed:
Update readme on how to build standalone tests

agunashe commented on projects/hip-tests/catch/README.md:
```diff
@@ -169,14 +169,73 @@ REQUIRE(exepctedOutput == proc.getOutput()); // Test on expected output of the p
+
+Install step, you might need to have superuser permission to be able to install globally.
+
+- `make install`
```
need a comment to mention where the built catch will be installed.
I suppose its /usr/local/lib ?

cjatin replied:
The default installation location is `/usr/local/`. Will update the readme.

jungx098 replied:
862dc0c63e

agunashe commented on projects/hip-tests/catch/README.md:
````diff
@@ -169,14 +169,73 @@ REQUIRE(exepctedOutput == proc.getOutput()); // Test on expected output of the p
+
 ```bash
-hipcc <path_to_test.cpp> -I<HIP_SRC_DIR>/tests/catch/include <HIP_SRC_DIR>/tests/catch/hipTestMain/standalone_main.cc -I<HIP_SRC_DIR>/tests/catch/external/Catch2 -g -o <out_file_name>
+export ROCM_SYSTEMS_DIR=<path_to_rocm_systems>
````
I need to try on windows to see if these work.  But I can fix this Readme later with updated instructions

cjatin replied:
Yeah windows might require a different set of instructions (with some modifications)

jungx098 replied:
Pretty outdated. Let me know if you want to revisit; otherwise, I will go ahead and close this.

cjatin committed:
update default location

cjatin committed:
Update build instructions

cjatin committed:
Simplify the steps to build standalone tests

cjatin committed:
update use of compiler

manocharahul commented on projects/hip-tests/catch/README.md:
````diff
@@ -169,13 +180,85 @@ REQUIRE(exepctedOutput == proc.getOutput()); // Test on expected output of the p
+If you want to use `main` provided by hip-tests, use the following command:
+
+```bash
+amdclang++                                                                           \
````
Instead of using amdclang++ , should we mention usage of hipcc to compile tests ? 
`hipcc <test_name> catch/hipTestMain/main.cc -I catch/include/ -I build/_deps/catch2-src/src/ -I build/_deps/catch2-build/generated-includes/  -lCatch2 -L build/_deps/catch2-build/src/ -o <test_out> catch/hipTestMain/hip_test_context.cc  -I catch/external/picojson/`

cjatin replied:
I can mention the section for hipcc, it still compiles with it, but for tests on CI are now being compiled with `amdclang`. hipcc adds some flags on its own like `-O3`, by default. I think we should compile it with amdclang for better compatibility/reproduce-ability with the CI.

jungx098 replied:
Pretty outdated. Let me know if you want to revisit; otherwise, I will go ahead and close this.

cjatin committed:
Update readme

agunashe committed:
Merge remote-tracking branch 'origin/develop' into users/jatin/tests_readme

agunashe committed:
Readme for building hip-tests/catch test set and standalone tests

jujiang-del commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,159 @@
+- `make -j8`
+
+Install step, you might need to have superuser permission to be able to install globally.
+The default install location for linux is `/usr/local/`.
```
```suggestion
During installation, you may need superuser privileges to perform a global install. On Linux systems, the default installation path is `/usr/local/`.
```

jujiang-del commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,159 @@
+Starting with, it is no longer a single header, it is now a library, which needs to be linked to the test exe.
+
+If you are on your personal machine, it is highly recommended to install Catch2v3 (v3.8.1) locally on your system.
+This helps skip the download/build part in hip-tests and results in faster builds overall.
```
```suggestion
`standalone_main.cc`, which was previously used, has been removed. We now rely on the main function provided by Catch2 to build standalone tests. The test suite has been upgraded to Catch2 v3 (specifically v3.8.1). This transition from v2 to v3 introduced several significant changes in how Catch2 integrates with hip-tests. Most importantly, Catch2 is no longer a single-header library; it is now a compiled library that must be linked into the test executable.

For developers working on personal machines, installing Catch2 v3 (v3.8.1) system-wide is strongly recommended. This avoids the download and build steps within hip-tests and leads to noticeably faster build times.

```

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,159 @@
+  -L /usr/local/lib -lCatch2 -lCatch2Main                                            \
+  -x hip <path_to_test>
+```
+The command above builds hip without the `main` from hip-tests, if the purpose is just to run the test with as simple commands as possible, go with this.
````
```suggestion
The command above builds HIP without the `main` provided by hip-tests. If your goal is simply to run the tests using the most straightforward commands, this is the option to go with.
```

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,159 @@
+export DEPS_PATH=$PWD/_deps
+```
+
+We now use the Catch2 we just build to link to the tests.
````
```suggestion
We now use the Catch2 library we just built to link against the tests.
```

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,159 @@
+  -x hip <test_file.cc>
+```
+
+You might need to set `LD_LIBRARY_PATH` to be Catch2 location.
````
```suggestion
You may need to set `LD_LIBRARY_PATH` to point to Catch2 location.
```

jujiang-del reviewed (commented):
Please see comments.

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme

jungx098 committed:
Update projects/hip-tests/README.md

Co-authored-by: Julia Jiang <56359287+jujiang-del@users.noreply.github.com>

jungx098 committed:
Update projects/hip-tests/README.md

Co-authored-by: Julia Jiang <56359287+jujiang-del@users.noreply.github.com>

jungx098 committed:
Update projects/hip-tests/README.md

Co-authored-by: Julia Jiang <56359287+jujiang-del@users.noreply.github.com>

jungx098 committed:
Update projects/hip-tests/README.md

Co-authored-by: Julia Jiang <56359287+jujiang-del@users.noreply.github.com>

jungx098 committed:
Apply suggestions from code review

Co-authored-by: Julia Jiang <56359287+jujiang-del@users.noreply.github.com>

jungx098 committed:
docs: Update HIP catch test README with macro docs

- Document HIP_TEST_CASE and HIP_TEMPLATE_TEST_CASE macros
  with usage examples and YAML config workflow
- Fix typos and grammar throughout (occurred, validated,
  it's, API capitalization, broken words)
- Clean up trailing whitespace and formatting issues

jungx098 committed:
docs: Restructure HIP catch test build docs

- Reorganize Catch2 setup into system-wide install
  and FetchContent sections with standalone commands
- Add build examples for with/without hip-tests main
  paired with each Catch2 install method
- Fix typos, formatting issues, and update copyright
  year to 2026

jungx098 reviewed (approved):
LGTM.

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+$ make -j32 build_tests
+$ ctest # run tests
+```
+
````
Need to put note after the build command:
Note: Developers can run `rocminfo` to determine the gfxXXXX architecture from the agent name. For example:
```bash
$ /opt/rocm/bin/rocminfo
...
*******                  
Agent 2                  
*******                  
  Name:                    gfx1200
```

Besides, if the HIP compiler cannot build even a simple test program, developers may need to explicitly set the paths for `amdclang` and `amdclang++` to point to `/opt/rocm/bin`.

jungx098 replied:
> Besides, if the HIP compiler cannot build even a simple test program, developers may need to explicitly set the paths for amdclang and amdclang++ to point to /opt/rocm/bin.

Could you clarify? I thought the HIP compiler can build a simple test, and `amdclang`/`amdclang++` come with ROCm (same path as `hipcc`).

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR50

jujiang-del commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,216 @@
+- `cmake .. -DCMAKE_BUILD_TYPE=Release`
+- `make -j8`
+
+During installation, you may need superuser privileges to perform a global install. On Linux systems, the default installation path is `/usr/local/`.
```
This sentence should appear immediately after the `sudo make install` command, formatted as follows:

Note: On Linux systems, the default installation path is /usr/local/.

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR83

jujiang-del commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,216 @@
+export ROCM_SYSTEMS_DIR=$PWD
+mkdir tests
+cd tests
+cmake ../projects/hip-tests/catch -DHIP_PLATFORM=amd -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=$HIP_PATH
```
I met build errors using this command like below:
CMake Error at CMakeLists.txt:285 (string):
  string sub-command REPLACE requires at least four arguments.
Please double check on this.

jungx098 replied:
Can you try this:
```sh
$ export ROCM_SYSTEMS_DIR=$PWD
$ export HIP_PATH=/opt/rocm
$ mkdir tests
$ cd tests
$ hipconfig --version
$ cmake ../projects/hip-tests/catch -DHIP_PLATFORM=amd -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=$HIP_PATH
```

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR96

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+  -L /usr/local/lib -lCatch2 -lCatch2Main                                            \
+  -x hip <path_to_test>
+```
+
````
I couldn't run this command.
besides, <path_to_test> is not clear here, and got error: error: error reading './tests': Is a directory

jungx098 replied:
<path_to_test> is a test case file, eg. `projects/hip-tests/catch/unit/gl_interop/hipGLGetDevices.cc`.  Looks like this doc misses `Catch2Main`.

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR142

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+  -lCatch2                                                                           \
+  -x hip <path_to_test>
+```
+
````
please double check this command. especially what needs to put as <path_to_test>

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR142

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+  -lCatch2                                                                           \
+  -x hip <path_to_test>
+```
+
````
please double check this command. especially what needs to put as <path_to_test>

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR142

jujiang-del commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+  -lCatch2                                                                           \
+  -x hip <path_to_test>
+```
+
````
please double check this command. especially what needs to put as <path_to_test>

jungx098 replied:
https://github.com/ROCm/rocm-systems/pull/2198/changes/004e7950c39f5bab0ecaa5f58555dd9dcd3b86ef#diff-adde73363fcf0710ccf0551f3670dfe1cd8c9422eb39cb930e6700d6df659f6dR142

jujiang-del reviewed (commented):
see comments.

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme

jungx098 commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,216 @@
+$ cd "$HIP_TESTS_DIR"
+$ mkdir -p build; cd build
+$ cmake ../catch -DHIP_PLATFORM=amd -DCMAKE_PREFIX_PATH=<HIP-Installed-Path> -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=amdclang++ -DCMAKE_C_COMPILER=amdclang -DCMAKE_HIP_COMPILER=amdclang++ -DOFFLOAD_ARCH_STR="--offload-arch=gfxXXXX"
+$ make -j32 build_tests
```
Use `-j$(nproc)` for consistency.

jungx098 commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,216 @@
+If you prefer not to install Catch2 system-wide, you can build it via cmake's FetchContent mechanism. The hip-tests cmake will automatically download and build Catch2 if it is not found on the system. To build just the Catch2 library:
+
+```bash
+export HIP_PATH=<path_where_hip_is_installed>
````
Inconsistent with others; they use `HIP_DIR`.

jungx098 replied:
HIP_DIR is used for the source code path.

jungx098 commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,216 @@
+mkdir tests
+cd tests
+cmake ../projects/hip-tests/catch -DHIP_PLATFORM=amd -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=$HIP_PATH
+make -j$(nproc) Catch2
```
It looks like, `make -j Catch2WithMain` is missing here.

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme

jungx098 committed:
docs: Update hip-tests README with build guidance

- Replace fixed -j flags with nproc; add rocminfo steps for
  --offload-arch and compiler PATH notes
- Update FetchContent flow: HIP_PATH, Catch2WithMain target, sudo
  install, and LD_LIBRARY_PATH note
- Clarify standalone compilation for system Catch2 vs FetchContent
  and add example test path

jungx098 commented on projects/hip-tests/README.md:
````diff
@@ -1,6 +1,235 @@
+
+Compile one test `.cc` with Catch2 supplying `main`. The hip-tests test runner and its shared context are not used.
+
+**System-wide Catch2** — link `Catch2` and `Catch2Main` (Catch2’s `main`):
+
+```bash
+amdclang++                                                                           \
+  -I $ROCM_SYSTEMS_DIR/projects/hip-tests/catch/include                              \
+  --offload-arch=native                                                              \
+  -L /usr/local/lib -lCatch2 -lCatch2Main                                            \
````
This feedback appears to be invalid and based solely on this file’s content. `Catch2WithMain` produces `Catch2.so` and `Catch2Main.so`.

jungx098 commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,235 @@
+  -I <build_dir>                                                                     \
+  -I $ROCM_SYSTEMS_DIR/projects/hip-tests/catch/include                              \
+  --offload-arch=native                                                              \
+  -L /usr/local/lib -lCatch2 -lCatch2Main                                            \
```
These standalone build instructions reference linking against -lCatch2Main, but hip-tests' CMake links against Catch2::Catch2WithMain for Catch2 v3. Please update the documented library name(s) to match Catch2 v3 (e.g., Catch2WithMain), otherwise the command is likely to fail on systems where Catch2Main is not produced/installed.

jungx098 committed:
docs: Fix encoding artifacts in hip-tests README

- Replace malformed quote characters with standard ASCII quotes
  in disclaimer
- Fix missing spaces after sentence-ending periods

jungx098 commented on projects/hip-tests/README.md:
```diff
@@ -1,6 +1,235 @@
+This repository provides unit tests for [HIP](../hip) implementation.
 
-This repository provides unit tests for  [HIP](https://github.com/ROCm/rocm-systems/tree/develop/projects/hip) implementation.
+## DISCLAIMER
+
+The information presented in this document is for informational purposes only and may contain technical inaccuracies, omissions, and typographical errors. The information contained herein is subject to change and may be rendered inaccurate for many reasons, including but not limited to product and roadmap changes, component and motherboard version changes, new model and/or product releases, produ ...
+
+©2026 Advanced Micro Devices, Inc. All Rights Reserved.
```
https://github.com/ROCm/rocm-systems/pull/2198/commits/2a9b45c4aa61f72e8a5b6ba70fd919d7a1f0099e

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme

jungx098 committed:
Merge branch 'develop' into users/jatin/tests_readme
